algoliasearch/transport.go

```diff
@@ -95,8 +95,11 @@ func NewTransportWithHosts(appID, apiKey string, hosts []string) *Transport {
 
 // setTimeout changes the timeouts used by the underlying HTTP client.
 func (t *Transport) setTimeout(connectTimeout, readTimeout time.Duration) {
-	t.httpClient.Transport.(*http.Transport).TLSHandshakeTimeout = connectTimeout
-	t.httpClient.Transport.(*http.Transport).ResponseHeaderTimeout = readTimeout
+	switch transport := t.httpClient.Transport.(type) {
+	case *http.Transport:
+		transport.TLSHandshakeTimeout = connectTimeout
+		transport.ResponseHeaderTimeout = readTimeout
+	}
```

Comment:
Add a `default` case displaying an error message (`Cannot use setTimeout for the underlying Transport` for instance).